resources/js/tableofcontent.js

```diff
@@ -72,4 +72,10 @@ function tocJSON()
 new QWebChannel(qt.webChannelTransport, function(channel) {
     var kiwixObj = channel.objects.kiwixChannelObj;
     kiwixObj.sendTableOfContent(tocJSON());
+    kiwixObj.navigationRequested.connect(function(anchor) {
+        var anchorIndex = anchor.lastIndexOf('#');
+        var anchorPageUrl = anchor.substr(0, anchorIndex);
+        if (window.location.href.replace(location.hash,"") === anchorPageUrl)
+            window.location.href = anchor;
+    })
```

Comment:
This comment is about the full commit - can you elaborate on the scenario that is problematic if the navigation is implemented in C++? Is your solution fully devoid of similar problems or they are just made less embarrassing for the UX?

Reply:
Ok, I thought I replaced this commit already... I previously forgot the setUrl method and was using this method. Another victim from rebasing.

Reply:
Done